src/app/utils/tests/RulesTestUtils.spec.ts

```diff
@@ -0,0 +1,32 @@
+import { AbstractGameState } from 'src/app/jscaip/GameState';
+import { LegalityStatus } from 'src/app/jscaip/LegalityStatus';
+import { Move } from 'src/app/jscaip/Move';
+import { Rules } from 'src/app/jscaip/Rules';
+
+export class RulesUtils {
+
+    public static expectMoveSuccess(rules: Rules<Move, AbstractGameState>,
+                                    state: AbstractGameState,
+                                    move: Move,
+                                    expectedState: AbstractGameState)
+    : void
+    {
+        const legality: LegalityStatus = rules.isLegal(move, state);
+        expect(legality.legal).toBeTruthy();
+        if (legality.legal.isSuccess()) {
+            const resultingState: AbstractGameState = rules.applyLegalMove(move, state, legality);
+            expect(resultingState.equals(expectedState)).withContext('state should be equals').toBeTrue();
```

Comment:
Par quoi suggère tu de remplacer cette ligne si les GameState sont in-fine pas comparable ?

Reply:
`expect(resultingState).toEqual(expectedState)` si on décide de pas garder le comparable pour les state

Reply:
yeup simplement oui !
Plus qu'à utiliser RulesUtils.expectMoveSuccess partout maintenant :D ! J'me sent bien d'avoir fini ça d'ici 21 tickets !